static/js/components/PaymentHistory.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+/* global SETTINGS: false */
+import React from 'react';
+import R from 'ramda';
+
+import { formatDollarAmount } from '../util/util';
+import type { RestState } from '../rest';
+
+export default class PaymentHistory extends React.Component {
+  props: {
+    klasses: RestState
+  };
+
+  renderPaymentRow = (klass: Object) => {
+    let paymentAmountMsg = klass.total_paid !== klass.price ?
+      `${formatDollarAmount(klass.total_paid)} out of ${formatDollarAmount(klass.price)}` :
+      `${formatDollarAmount(klass.total_paid)}`;
+
+    return <tr key={klass.klass_key}>
+      <td>{klass.klass_name}</td>
+      <td>{paymentAmountMsg}</td>
+      <td className="statement-column">
+        <i className="material-icons">print</i><a href="#" className="statement-link">View Statement</a>
+      </td>
+    </tr>;
+  };
+
+  renderPaymentRows = () => {
+    const { klasses } = this.props;
+
+    return R.pipe(
+      R.filter(
+        R.compose(
+          R.not,
+          R.equals(0),
+          R.propOr(0, 'total_paid')
+        )
+      ),
+      R.map(this.renderPaymentRow)
```

Comment:
It looks like `renderPaymentRow` is a pure function (it does not depend on anything other than it's argument for calculating its return value) so it might be a good idea to pull `renderPaymentRow` and this `R.pipe` call out as module-level functions, so we can avoid re-declaring this function object every time we call `renderPaymentRows`.